buildman/manager/ephemeral.py

```diff
@@ -41,271 +58,114 @@
-                "Terminating expired build executor for job %s with execution id %s",
-                build_job.build_uuid,
-                build_info.execution_id,
-            )
-            await self.kill_builder_executor(build_job.build_uuid)
-        else:
-            logger.warning(
-                "Unexpected KeyEvent (%s) on job key: %s", key_change.event, key_change.key
-            )
-
-    async def _realm_callback(self, key_change):
-        logger.debug("realm callback for key: %s", key_change.key)
-        if key_change.event == KeyEvent.CREATE:
-            # Listen on the realm created by ourselves or another worker.
-            realm_spec = json.loads(key_change.value)
-            self._register_realm(realm_spec)
-
-        elif key_change.event in (KeyEvent.DELETE, KeyEvent.EXPIRE):
-            # Stop listening for new connections on the realm, if we did not get the connection.
-            realm_spec = json.loads(key_change.value)
-            realm_id = realm_spec["realm"]
-
-            build_job = BuildJob(AttrDict(realm_spec["job_queue_item"]))
-            build_uuid = build_job.build_uuid
-
-            logger.debug("Realm key %s for build %s was %s", realm_id, build_uuid, key_change.event)
-            build_info = self._build_uuid_to_info.get(build_uuid, None)
-            if build_info is not None:
-                # Pop off the component and if we find one, then the build has not connected to this
-                # manager, so we can safely unregister its component.
-                component = self._component_to_job.pop(build_info.component, None)
-                if component is not None:
-                    # We were not the manager which the worker connected to, remove the bookkeeping for it
-                    logger.debug("Unregistering unused component for build %s", build_uuid)
-                    self.unregister_component(build_info.component)
-
-            # If the realm has expired, then perform cleanup of the executor.
-            if key_change.event == KeyEvent.EXPIRE:
-                execution_id = realm_spec.get("execution_id", None)
-                executor_name = realm_spec.get("executor_name", "EC2Executor")
-
-                # Cleanup the job, since it never started.
-                logger.debug("Job %s for incomplete marking: %s", build_uuid, build_info)
-                if build_info is not None:
-                    await self._mark_job_incomplete(build_job, build_info)
-
-                # Cleanup the executor.
-                logger.debug(
-                    "Realm %s expired for job %s, terminating executor %s with execution id %s",
-                    realm_id,
-                    build_uuid,
-                    executor_name,
-                    execution_id,
-                )
-                await self.terminate_executor(executor_name, execution_id)
+        self._manager_config = {}
+        self._orchestrator = None
 
+    def initialize(self, manager_config):
```

Comment:
This might be a dumb question, but why not `initialize()` in `__init__()`? I don't see that method defined in the ABC.